testing/unittests/jlibtests.cpp

```diff
@@ -3396,7 +3397,629 @@ element1: !el scalarvalue
+              - name: secret-if-a
+                presence: p
+                maskStyle:
+                - name: keep-last-4-numbers
+                  presence: r
+                - name: mask-last-4-numbers
+                  presence: o
+                Set:
+                - name: value-type-set-a
+                  presence: r
+              - name: secret-if-b
+                presence: r
+                Set:
+                - name: value-type-set-b
+                  presence: r
+              rule:
+              - contentType: ''
+                presence: r
+              - contentType: xml
+                presence: r
+          - compatibility:
+              context:
+                domain: 'urn:hpcc:unittest'
+                version: 0
+                property:
+                - name: valuetype-set
+                - name: rule-set
+                  value: 'rule-set-2'
+              valueType:
+              - name: secret
+                presence: r
+              - name: secret-if-a
+                presence: p
+                maskStyle:
+                - name: keep-last-4-numbers
+                  presence: r
+                - name: mask-last-4-numbers
+                  presence: o
+                Set:
+                - name: value-type-set-a
+                  presence: r
+              - name: secret-if-b
+                presence: p
+                Set:
+                - name: value-type-set-b
+                  presence: r
+              rule:
+              - contentType: ''
+                presence: p
+              - contentType: xml
+                presence: p
+      )!!!";
+    Owned<IPropertyTree> originalTree;
+
+public:
+    void testSerializeVsSerializeToStreamRootOnlyPTree()
+    {
+        testSerializeVsSerializeToStream(createPTree("EmptyRoot"));
+    }
+
+    void testSerializeVsSerializeToStreamFromXmlPTree()
+    {
+        testSerializeVsSerializeToStream(createPTree(testXml));
+    }
+
+    void testSerializeVsSerializeToStreamFromYamlPTree()
+    {
+        testSerializeVsSerializeToStream(createPTreeFromYAMLString(chkTxt));
+    }
+
+    void testSerializeVsSerializeToStreamCPTValuePTree()
+    {
+        testSerializeVsSerializeToStream(createCPTValuePTree());
+    }
+
+protected:
+    void testSerializeVsSerializeToStream(Owned<IPropertyTree> _originalTree)
+    {
+        originalTree.setown(_originalTree.getClear());
+
+        MemoryBuffer memBuf;
+        originalTree->serialize(memBuf);
+
+        StringBuffer stringStreamBuf;
+        Owned<IBufferedSerialOutputStream> out = createBufferedSerialOutputStream(stringStreamBuf);
+        originalTree->serializeToStream(*out);
+        out->flush();
+        MemoryBuffer memBufStream;
+        memBufStream.append(stringStreamBuf.length(), stringStreamBuf);
+        memBufStream.reset(0);
+        Owned<IBufferedSerialInputStream> in = createMemoryBufferSerialStream(memBufStream);
+        CPPUNIT_ASSERT(in);
+
```

Comment:
It would be interesting to time each of the calls, and log at the end.  (Will come out if -v is specified on the unit test).